packages/react/src/hooks/accounts/useConnect.ts

```diff
@@ -70,14 +73,38 @@ export function useConnect({
   }, [client, forceUpdate])
 
   const connect = React.useCallback(
-    (connector_?: ConnectArgs['connector']) =>
-      mutate(<ConnectArgs>{ connector: connector_ ?? connector }),
+    (connectorOrArgs?: Partial<ConnectArgs> | ConnectArgs['connector']) => {
+      if (connectorOrArgs instanceof Connector) {
+        const connector_ = connectorOrArgs
+        return mutate(<ConnectArgs>{
+          chainId,
+          connector: connector_ ?? connector,
+        })
+      }
+      const args = connectorOrArgs
+      return mutate(<ConnectArgs>{
+        chainId: args?.chainId ?? chainId,
+        connector: args?.connector ?? connector,
+      })
+    },
```

Comment:
Think it's easier to construct these args and pass to `mutate` once or construct independently and call twice?